src/block_compressor.rs

```diff
@@ -63,7 +63,7 @@ pub struct GpuBlockCompressor {
 
 impl GpuBlockCompressor {
     /// Creates a new block compressor instance.
-    pub fn new(device: Arc<Device>, queue: Arc<Queue>) -> Self {
+    pub fn new(device: Device, queue: Queue) -> Self {
```

Comment:
I think we should explain in the comment, that the device and queue can safely be cloned, since they are internally reference counted by WGPU itself.

Reply:
Do you mean the doc comment? Another solution is to take references in argument and clone them ourselves.

Reply:
I also though about using references here, but then this kinda would lead the caller to believe, that we don't take ownership of the device and queue. I think it would best to extend the doc comment instead.